File: VB/ConsoleApp/Printers/ConsolePrinter.vb
Namespace Printers

    Public Class ConsolePrinter
        Implements IPrint

        Private _collection As New List(Of IPrintable)
        Public ReadOnly Property Collection As List(Of IPrintable)
            Get
                Return _collection
            End Get
        End Property

        Public Function Print() As String Implements IPrint.Print
            Dim sb As New System.Text.StringBuilder()

            sb.AppendLine("----------------------------------------------------------------")
            sb.AppendLine("Current Drawing")
            sb.AppendLine("----------------------------------------------------------------")

            For Each element In Collection
                sb.AppendLine(element.ToString())
            Next

            sb.AppendLine("----------------------------------------------------------------")

            Return sb.ToString()
        End Function

    End Class

End Namespace

File: CSharp/ConsoleApp/ConsoleApp/Concrete/ConsolePrinter.cs
using ConsoleApp.Abstractions;
using System.Collections.Generic;
using System.Text;

namespace ConsoleApp.Concrete
{
    public class ConsolePrinter : IPrinter
    {
        private readonly IDrawing _drawing;
        public IDrawing Drawing
        {
            get
            {
                return _drawing;
            }
        }

        public ConsolePrinter(IDrawing drawing)
        {
            _drawing = drawing;
        }

        public string Print()
        {
            var sb = new StringBuilder();

            sb.AppendLine("----------------------------------------------------------------");
            sb.AppendLine("Current Drawing");
            sb.AppendLine("----------------------------------------------------------------");

            var items = (List<IPrint>)_drawing;

            foreach (var element in items)
            {
                sb.AppendLine(element.ToString());
            }

            sb.AppendLine("----------------------------------------------------------------");

            return sb.ToString();
        }

    }

}
